mllib/src/main/scala/org/apache/spark/mllib/tree/impl/NodeIdCache.scala

```diff
@@ -131,7 +131,7 @@ private[spark] class NodeIdCache(
     rddUpdateCount += 1
 
     // Handle checkpointing if the directory is not None.
-    if (nodeIdsForInstances.sparkContext.getCheckpointDir.nonEmpty &&
+    if (nodeIdsForInstances.sparkContext.getCheckpointDir.nonEmpty && checkpointInterval != -1 &&
```

Comment:
This should be changed in the spark.ml version of NodeIdCache.  (I know this is confusing having two copies of it.  I'm planning on removing the old one during this release.)